app/controllers/user_registrations_controller.rb

```diff
@@ -8,22 +8,22 @@ class UserRegistrationsController < Spree::UserRegistrationsController
   # POST /resource/sign_up
   def create
     @user = build_resource(params[:spree_user])
-    if resource.save
-      session[:spree_user_signup] = true
-      session[:confirmation_return_url] = params[:return_url]
-      associate_user
+    unless resource.save
+      return render_error(@user.errors)
```

Comment:
This could be a one-liner now:
```rb
return render_error(@user.errors) unless resource.save
```

`resource` and `@user` are the same, right?

Reply:
sorry @mkllnk I forgot this comment...
yes, could be a one liner. but I also like this pattern because the one liner moves this major point in the happy path of the method (resource.save) to the end of the line...

yes, I think resource and @user are the same but I didnt want to touch it. anyway this assignment to @user and then usage of resource in the next line is very very ugly...

because this is already ready to go I'll skip changing these things.